### DISCUSSION TITLE
Change WatchProgram constructor args

### DISCUSSION THREAD

**Discussion by draylegend** [phero-hq/phero#105] [Ideas]
I wanna develop an nx-plugin that initializes and serves the phero server and client. In order to get the plugin to work I need to pass two files like this:

```json
// phero-api/project.json
{
  "targets": {
    "serve": {
      "executor": "@draylegend/phero-server:serve", // example executor's name
      "options": {
        "tsConfig": "apps/phero-api/tsconfig.app.json", // tsconfig.app.json is a special config file, used by nx/angular
        "phero": "apps/phero-api/src/phero.ts"
      }
    }
  // ...
}
```

I'd suggest to extend the args of WatchProgram constructor. I'd give the ability to change the paths to `tsconfig.json` and `phero.ts`.

```ts
interface ServerCommandServe {
    name: ServerCommandName.Serve;
    tsConfig: string; // e.g. d:\coffee-shop\apps\phero-api\tsconfig.lib.json
    phero: string; // e.g. d:\coffee-shop\apps\phero-api\src\phero.ts
    verbose: boolean;
    port: number;
}

// command: ServerCommandServe
new DevServer(command);

new WatchProgram(
  tsConfig,
  phero,
  //..
);
```

Usage could look like this: `phero-server serve --tsConfig=apps/phero-api/tsconfig.app.json --phero=apps/phero-api/src/phero.ts`
or without additional args `phero-server serve`.

I already adapted the `phero-server` cli to suggested options. If this is worth the effort, I'd try to create an nx-plugin with `init` and `serve` commands.

P.S. I have a branch created locally, but I can't push this branch to github.

**Comment by JasperH8g** [comment]
I'm not very familiar with `nx` myself yet, but we'll dive in to properly understand this. Two things:

- Why exactly would you need a separate tsconfig-file for phero? 
- We're working on an iteration of the compiler, with which will find any `phero.ts` file in your codebase and take anything in it to generate the manifest and client. Would you still need an option to point to the phero-file specifically?

**Reply by draylegend** [reply]
I'll wait till the `phero.ts`-lookup functionality is done. This is even a better approach, than passing as a cli argument
